identity-service/src/app.js

```diff
@@ -329,6 +329,19 @@ class App {
     return [listenCountLimiter, listenCountIPLimiter]
   }
 
+  // Create a rate limiter for listens exclusively based on IP
+  _createDailyRateLimitForListenCountIP () {
```

Comment:
we may want to have hour & week values for this, so can we use the above structure?

```
 // Create rate limits for listens on a per track per user basis and per track per ip basis
  _createRateLimitsForListenCounts (interval, timeInSeconds) {
    const listenCountLimiter = getRateLimiter({
      prefix: `listenCountLimiter:::${interval}-track:::`,
      expiry: timeInSeconds,
      max: config.get(`rateLimitingListensPerTrackPer${interval}`), // max requests per interval
      skip: function (req) {
        const { ip, senderIP } = getIP(req)
        const ipToCheck = senderIP || ip
        // Do not apply user-specific rate limits for any whitelisted IP
        return isIPWhitelisted(ipToCheck, req)
      },
      keyGenerator: function (req) {
        const trackId = req.params.id
        const userId = req.body.userId
        return `${trackId}:::${userId}`
      }
    })
    const listenCountIPTrackLimiter = getRateLimiter({
      prefix: `listenCountLimiter:::${interval}-ip-track:::`,
      expiry: timeInSeconds,
      max: config.get(`rateLimitingListensPerIPPer${interval}`), // max requests per interval
      keyGenerator: function (req) {
        const trackId = req.params.id
        const { ip } = getIP(req)
        return `${ip}:::${trackId}`
      }
    })
   const listenCountIPLimiter = getRateLimiter({
      prefix: `listenCountLimiter:::${interval}-ip:::`,
      expiry: timeInSeconds,
      max: config.get(`rateLimitingListensExclusivelyPerIPPer${interval}`), // max requests per day
      keyGenerator: function (req) {
        const { ip } = getIP(req)
        return `${ip}`
      }
    })
    return [listenCountLimiter, listenCountIPLimiter, listenCountIPTrackLimiter]
  }

```

Reply:
below, we could then just do

const [_, _, listenCountDailyIPLimiter] = this._createRateLImitsForListenCounts('Day', ONE_HOUR_IN_SECONDS * 24)

Reply:
yeah this is much better, will make the change